c10/util/safe_numerics.h

```diff
@@ -22,7 +22,13 @@ C10_ALWAYS_INLINE bool add_overflows(uint64_t a, uint64_t b, uint64_t* out) {
   return __builtin_add_overflow(a, b, out);
 #else
   unsigned long long tmp;
-  auto carry = _addcarry_u64(0, a, b, &tmp);
+  #if defined (_M_ARM64) || defined (_M_HYBRID_X86_ARM64) || defined (_M_ARM64EC)
+    tmp = a + b;
+    unsigned long long vector = (a & b) ^ ((a ^ b) & ~tmp);
+    auto carry = vector >> 63;
+  #else
+    auto carry = _addcarry_u64(0, a, b, &tmp);
+  #endif
```

Comment:
Can this check be inverted to use `_addcarry_u64` instrinsic if compiled for x86_64 platform otherwise use explicit implementation, seomthing like:
```
#if defined(_M_X64) || defined(__x86_64__)
  auto carry = _addcarry_u64(0, a, b, &tmp);
#else
  tmp = a + b;
  ...
#endif

Reply:
That makes sense, I'll add a commit to this PR ASAP to use software approach as default fallback.